main.py

```
@@ -0,0 +1,179 @@
+    search_client = discoveryengine.SearchServiceClient()
+except Exception as e:
+    print(f"Error al inicializar el cliente de Vertex AI Search: {e}")
+    raise HTTPException(
+        status_code=500, detail=f"Error al inicializar el cliente de búsqueda: {str(e)}"
+    )
+
+class QueryRequest(BaseModel):
+    query: str
+
+@app.post("/query")
+async def query(request: QueryRequest):
```

Ruff (0.12.2)

`query` is too complex (18 > 10)

(C901)